webextension/privileged/ui/CloudDownloadsView.jsm

```diff
@@ -230,16 +227,46 @@ var CloudDownloadsView = {
     Services.wm.removeListener(WindowListener);
   },
 
+  // Log provider count and provider keys existing on user desktop
+  async handleAddOnInitTelemetry() {
+    // There is no other API that awaits the result of CloudStorageInternal.init()
+    // Invoke getDownloadFolder on CloudStorage API to ensure API is initialized
+    // before asking CloudStorage API for storage providers info
+    await CloudStorage.getDownloadFolder();
+    const providers = await CloudStorage.getStorageProviders();
+
+    const keys = [];
+    if (providers.size > 0) {
+      providers.forEach((value, key) => {
+        keys.push(key);
+      });
+    }
+
+    this.eventEmitter.emit("record-telemetry", {
+      message: "addon_init",
+      provider_count: providers.size.toString(),
+      provider_keys: keys.join(","),
+    });
+
+    this.providers = await this._filterProvidersMap(providers);
+  },
+
   async toggleAPIEnabledState() {
     try {
       if (!this.gIsAPIEnabled) {
         if (this.isInitialized) {
-          this.uninitWindowListener();
+          if (this.studyVariation !== "control") {
+            this.uninitWindowListener();
+          }
           this.unRegisterNotification();
         }
         return;
       }
-      await this.initWindowListener();
+
+      await this.handleAddOnInitTelemetry();
+      if (this.studyVariation !== "control") {
+        await this.initWindowListener();
+      }
       this.registerNotification();
```

Comment:
Why do we need to do the non-window-listener thing even in the control case?

Reply:
Control branch will not have any of cloud storage feature UI  which is setup inside initWindowListener.  To compare  if there is difference in downloads with or without cloud storage feature, for all branches we are logging
a) how many download triggered by the user (download_started)
https://github.com/mozilla/shield-cloudstorage/pull/28/commits/2d9d464fc91c5fbdbb631ca8c3fdaf7c705cd5c9#diff-7e1f5d4fd67bdee7405adbadd873d06bR279

b) Number of cloud storage providers (addon_init inside handleAddOnInitTelemetry)
https://github.com/mozilla/shield-cloudstorage/pull/28/commits/2d9d464fc91c5fbdbb631ca8c3fdaf7c705cd5c9#diff-7e1f5d4fd67bdee7405adbadd873d06bR266

Rest of the telemetry documented inside /docs/telemetry.json, will be only for experimental branches showing cloud storage feature UI

Will update docs with branches tested in the study

Reply:
@gijsk About moving initialization of providers inside handleAddOnInitTelemetry, we are doing so to invoke this method for all branches including control.

Moving it outside of initWindowListener gives us an additional benefit to initialize providers only one time if multiple windows are open.